### PR TITLE
Add Zig support

### LINKED ISSUE #3724: [Feature request] Zig Language Support
Hello!

This is just a request to add the [Zig language](https://ziglang.org/) support (syntax highlight, file type choice (.zig), ability to build and execute, etc.)
I haven't seen a similar open issue, so decided to file one.

Thank you!

### PR DESCRIPTION
**Pull request by techee**
Uses the recently merged Zig lexer from lexilla.

Fixes #3724.

### CODE REVIEW

**Comment by eht16** [comment]
LGBI. Not that I know anything about Zig, just a bit of Geany :)

**Comment by techee** [comment]
@eht16 Thanks for having a look at it.

I forgot to mention that there hasn't been a formal lexilla release with Zig or Dart yet. This means that the changes in SciLexer.h for these lexers had to be added manually rather than using the real Lexilla's unmodified file. In any case, once Geany updates to the latest lexilla, it should be auto-fixed.

File: src/highlighting.c [hunk 1]
```diff
@@ -1685,6 +1687,10 @@ gboolean highlighting_is_string_style(gint lexer, gint style)
 				style == SCE_NIM_TRIPLE ||
 				style == SCE_NIM_TRIPLEDOUBLE ||
 				style == SCE_NIM_STRINGEOL);
+
+		case SCLEX_ZIG:
+			return (style == SCE_ZIG_STRING ||
+				style == SCE_ZIG_MULTISTRING);
```
Comment (b4n) [inline src/highlighting.c:1693]: shouldn't this include `CHARACTER` and `ESCAPECHAR`?  I didn't check the lexer yet

**Comment by elextr** [comment]
> I forgot to mention that there hasn't been a formal lexilla release with Zig or Dart yet. This means that the changes in SciLexer.h for these lexers had to be added manually rather than using the real Lexilla's unmodified file. In any case, once Geany updates to the latest lexilla, it should be auto-fixed.

Fine, we have done that before until Geany updates of Lexilla and ctags which is usually slower than Lexilla and ctags releases ;-)

File: src/highlighting.c (hunk 1 above)
Comment (techee) [reply src/highlighting.c]: Correct, done.

File: scintilla/lexilla/include/SciLexer.h [hunk 2]
```diff
@@ -2058,6 +2062,93 @@
...
+#define SCE_TOML_KEY 6
+#define SCE_TOML_ERROR 7
+#define SCE_TOML_OPERATOR 8
+#define SCE_TOML_STRING_SQ 9
+#define SCE_TOML_STRING_DQ 10
+#define SCE_TOML_TRIPLE_STRING_SQ 11
+#define SCE_TOML_TRIPLE_STRING_DQ 12
+#define SCE_TOML_ESCAPECHAR 13
+#define SCE_TOML_DATETIME 14
+#define SCE_TROFF_DEFAULT 0
```
Comment (elextr) [inline scintilla/lexilla/include/SciLexer.h:2080]: troff ??

Comment (techee) [reply scintilla/lexilla/include/SciLexer.h:2080]: Yes, this is `SciLexer.h` taken from lexilla - it contains all the styles for all the languages that lexilla supports (we don't patch it in any way). In addition to Zig, Dart, and TOML that I added to lexilla, the troff lexer was merged independently of me so there's this extra language as well.

Comment (elextr) [reply scintilla/lexilla/include/SciLexer.h:2080]: Ok

**Review by b4n — APPROVED** [review]
LGTM.  I don't know any Zig, but this PR seems to work OK.

@techee would you like to squash it a bit, or merge as-is?

**Comment by techee** [comment]
> @techee would you like to squash it a bit, or merge as-is?

Yeah, squashed. Merging. Thanks for having a look at it.

**Pull request merged**